pytorch_toolkit/object_detection/tests/common/test_case.py

```diff
@@ -260,3 +299,109 @@ def test_alt_ssd_export_on_cpu(self):
         return ExportWithAltSsdTestCase
 
     return ExportTestCase
+
+
+def create_nncf_test_case(problem_name, model_name, ann_file, img_root, template_update_dict):
+    """
+    Note that template_update_dict will be used to update template file
+    using the function mmcv.Config.merge_from_dict
+    """
+
+    assert template_update_dict, 'Use this function with non-trivial template_update_dict parameter only'
+    class TestCaseOteApi(unittest.TestCase):
+
+        @classmethod
+        def setUpClass(cls):
+            cls.template_updates_description = cls.generate_template_updates_description(template_update_dict)
+            logging.info(f'Begin setting up class for {problem_name}/{model_name}, {cls.template_updates_description}')
+
+            cls.templates_folder = os.environ['MODEL_TEMPLATES']
+            src_template_folder = os.path.join(cls.templates_folder, 'object_detection', problem_name, model_name)
+
+            cls.skip_non_instantiated_template_if_its_allowed(src_template_folder)
+            cls.src_template_folder = src_template_folder
+
+            cls.template_folder = cls.generate_template_folder_name(cls.src_template_folder,
+                                                                    cls.template_updates_description)
+            cls.copy_template_folder(cls.src_template_folder, cls.template_folder)
```

Comment:
what's for?

Reply:
Ilya, it is really required, since we need to make a lot of tests with one model.   

As an example, for `facedetection-0200`: we need to have
* a test on `facedetection-0200` with `int8` quantization
* a test on `facedetection-0200` with `sparsity` compression
* a test on `facedetection-0200` with `int8` quantization + `sparsity` compression
* etc

Since switchers for the compression types are placed in `template.yaml`, I need to change `template.yaml` for each of the test. Moreover, each of the test will store its temporary files (and results of training) in its model templates folder.
If we use one folder for all these tests, we will have a lot of tests and one will be run on the top of another.

As far as I know, the results of side effects between tests may be dangerous. So, I prefer to run each test in its own folder.

Moreover, @Ilya-Krylov, **I made one more change** -- see my commit fb6ad0b:    
now for NNCF tests the workdir will be in another folder than for all other tests.